x/posts/client/rest/tx.go

```diff
@@ -44,7 +44,16 @@ func createPostHandler(cliCtx context.CLIContext) http.HandlerFunc {
 			return
 		}
 
-		msg := types.NewMsgCreatePost(req.Message, parentID, req.AllowsComments, req.Subspace, req.OptionalData, addr, req.CreationTime)
+		var msg types.MsgCreatePost
+		msg = types.NewMsgCreateTextPost(req.Message, parentID, req.AllowsComments, req.Subspace, req.OptionalData,
+			addr, req.CreationTime)
+
+		if req.Medias != nil && len(req.Medias) != 0 {
```

Comment:
In Go, a `nil` array will have `len == 0` so the first check can be omitted without any problem, we can simply check for `len(req.Medias)`